### PR TITLE
Smooted's BM80v2 keymap + rules

### PR DESCRIPTION
**Pull request by Smooted**
### Smooted's Keymap + Added Rules for KPRepublic BM80v2 TKL 80% Keyboard

Added to Layer 1:

F6 - Mute Volume
F7 - Volume Down
F8 - Volume Up

-------------------------------------

Added VIA compatibility
Poll rate set to 1KHz

### CODE REVIEW

File: keyboards/kprepublic/bm80v2/keymaps/smooted/config.h [hunk 1]
```diff
@@ -0,0 +1,3 @@
+#pragma once
```
Comment (waffle87) [inline keyboards/kprepublic/bm80v2/keymaps/smooted/config.h:1]: This file is missing a license header.

File: keyboards/kprepublic/bm80v2/keymaps/smooted/config.h [hunk 2]
```diff
@@ -0,0 +1,3 @@
...
+#define FORCE_NKRO
+#define USB_POLLING_INTERVAL_MS 1
```
Comment (waffle87) [inline keyboards/kprepublic/bm80v2/keymaps/smooted/config.h:3]: This is the default behaviour.
```suggestion
```

File: keyboards/kprepublic/bm80v2/keymaps/smooted/keymap.c [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+    [3] = LAYOUT_tkl_ansi(
+        _______,          _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,    _______, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,    _______, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,    _______, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______,          _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,                      _______,
+        _______, _______, _______,                   _______,                                     _______, _______, _______, _______,    _______, _______, _______
+    )
+
```
Comment (waffle87) [inline keyboards/kprepublic/bm80v2/keymaps/smooted/keymap.c:42]:
```suggestion
    )
```

File: keyboards/kprepublic/bm80v2/keymaps/smooted/rules.mk [hunk 4]
```diff
@@ -0,0 +1,4 @@
+VIA_ENABLE = yes
+LTO_ENABLE = yes
+NKRO_ENABLE = yes
+
```
Comment (waffle87) [inline keyboards/kprepublic/bm80v2/keymaps/smooted/rules.mk:4]: These are already enabled at the keyboard level.
```suggestion
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

**Comment by Smooted** [comment]
> Just in case you weren't aware: VIA has a **Save + Load** feature, which will allow you to save customizations to the default layout to a file on your computer. You may then reload those customizations if they are ever lost.

Thank you, I'm actually extremely new to Github and QMK and all of this keyboard stuff. A friend told me to set this all up and it's all a little new and confusing to me so thank you for being patient lolz

**Comment by Smooted** [comment]
> > Thank you, I'm actually extremely new to Github and QMK and all of this keyboard stuff. A friend told me to set this all up and it's all a little new and confusing to me so thank you for being patient lolz
> 
> Gotcha. If you use Discord, you're welcome to stop by the QMK Discord if you have any questions.
> 
> The reason I mention that tidbit about VIA is because it means you don't actually have to submit your keymap to the main QMK repository, or even deal with GitHub at all if you just want to preserve the changes you've made to your VIA keymap.
> 
> Your call, of course. If you decide against it, [you can close this PR](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/closing-a-pull-request).

If I ever get around to getting this into the Github, I may just decide against VIA entirely and make edits on my local fork whenever I want to change key bindings, and explore what else I can do locally without bothering y'all too much. This QMK stuff is super cool. I'll double make sure this PR gets closed if you don't close it before me though. <3

**Comment by Smooted** [comment]
Did something fail during the build? I think in the keymap when it got to the matrix. Should I put it back to how it was? How I had it with those extra 2 layers (layer 2 and 3) is how the BM80 keymap is by default afaik.

File: keyboards/kprepublic/bm80v2/keymaps/smooted/keymap.c [hunk 5]
```diff
@@ -20,24 +20,4 @@ const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
...
-    [3] = LAYOUT_tkl_ansi(
-        _______,          _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,    _______, _______, _______,
-        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,    _______, _______, _______,
-        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,    _______, _______, _______,
-        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
-        _______,          _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,                      _______,
-        _______, _______, _______,                   _______,                                     _______, _______, _______, _______,    _______, _______, _______
     )
-
-};
```
Comment (waffle87) [inline keyboards/kprepublic/bm80v2/keymaps/smooted/keymap.c]: Removing this closing bracket is what's causing build errors.

**Review by waffle87 — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Pull request merged**